File: CS/dx_sample/XtraReport1.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using DevExpress.XtraReports.UI;
using DevExpress.XtraPrinting;
using System.Collections.Generic;
using System.Data;

namespace dx_sample {
    public partial class XtraReport1 : DevExpress.XtraReports.UI.XtraReport {
        Dictionary<string, float> sections = new Dictionary<string,float>();
        public XtraReport1() {
            InitializeComponent();
        }
        bool documentBuilt = false;
        private void xrSubreport1_BeforePrint(object sender, System.Drawing.Printing.PrintEventArgs e) {
            ((XRSubreport)sender).ReportSource.Parameters[0].Value = GetCurrentColumnValue("CustomerID").ToString();
        }
        public void CreateCustomDocument() {
            CreateDocument();
            documentBuilt = true;
            CreateDocument();
        }
        protected override void AfterReportPrint() {
            if (documentBuilt) return;
            foreach (DevExpress.XtraPrinting.Page page in Pages) {
                foreach (Brick brick in page) {
                    LabelBrick labelBrick = brick as LabelBrick;
                    if (labelBrick != null && labelBrick.Value != null && labelBrick.Value.ToString() != string.Empty) {
                        ((XRLabel)labelBrick.BrickOwner.RealControl).BackColor = Color.LightGreen;
                        if (sections.ContainsKey(labelBrick.Value.ToString()))
                            sections[labelBrick.Value.ToString()] += labelBrick.Size.Height;
                        else
                            sections.Add(labelBrick.Value.ToString(), labelBrick.Size.Height);
                    }
                }
            }
        }
        private void Detail_BeforePrint(object sender, System.Drawing.Printing.PrintEventArgs e) {
            string customerID = GetCurrentColumnValue("CustomerID").ToString();
            if (((XtraReport)xrSubreport1.ReportSource).Parameters.Count > 0)
                ((XtraReport) xrSubreport1.ReportSource).Parameters[0].Value = customerID;
            xrSubreport1.ReportSource.FillDataSource();
            xrSubreport1.ReportSource.ApplyFiltering();
            if (xrSubreport1.ReportSource.RowCount == 0) {
                e.Cancel = true;
                return;
            }
            if (!documentBuilt) return;
            if (!sections.ContainsKey(customerID))
                System.Diagnostics.Debug.WriteLine(customerID + " not found."); //  Not executed.
            else {
                xrLabel1.HeightF = 100 * GraphicsUnitConverter.Convert(sections[customerID], GraphicsUnit.Document, GraphicsUnit.Inch);
                ((DetailBand)sender).HeightF = 100 * GraphicsUnitConverter.Convert(sections[customerID], GraphicsUnit.Document, GraphicsUnit.Inch);
            }
        }
    }
}

File: VB/dx_sample/XtraReport1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI
Imports DevExpress.XtraPrinting
Imports System.Collections.Generic
Imports System.Data

Namespace dx_sample
	Partial Public Class XtraReport1
		Inherits DevExpress.XtraReports.UI.XtraReport
		Private sections As New Dictionary(Of String, Single)()
		Public Sub New()
			InitializeComponent()
		End Sub
		Private documentBuilt As Boolean = False
		Private Sub xrSubreport1_BeforePrint(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintEventArgs) Handles xrSubreport1.BeforePrint
			CType(sender, XRSubreport).ReportSource.Parameters(0).Value = GetCurrentColumnValue("CustomerID").ToString()
		End Sub
		Public Sub CreateCustomDocument()
			CreateDocument()
			documentBuilt = True
			CreateDocument()
		End Sub
		Protected Overrides Sub AfterReportPrint()
			If documentBuilt Then
				Return
			End If
			For Each page As DevExpress.XtraPrinting.Page In Pages
				For Each brick As Brick In page
					Dim labelBrick As LabelBrick = TryCast(brick, LabelBrick)
					If labelBrick IsNot Nothing AndAlso labelBrick.Value IsNot Nothing AndAlso labelBrick.Value.ToString() <> String.Empty Then
						CType(labelBrick.BrickOwner.RealControl, XRLabel).BackColor = Color.LightGreen
						If sections.ContainsKey(labelBrick.Value.ToString()) Then
							sections(labelBrick.Value.ToString()) += labelBrick.Size.Height
						Else
							sections.Add(labelBrick.Value.ToString(), labelBrick.Size.Height)
						End If
					End If
				Next brick
			Next page
		End Sub
		Private Sub Detail_BeforePrint(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintEventArgs) Handles Detail.BeforePrint
			Dim customerID As String = GetCurrentColumnValue("CustomerID").ToString()
			If (CType(xrSubreport1.ReportSource, XtraReport)).Parameters.Count > 0 Then
				CType(xrSubreport1.ReportSource, XtraReport).Parameters(0).Value = customerID
			End If
			xrSubreport1.ReportSource.FillDataSource()
			xrSubreport1.ReportSource.ApplyFiltering()
			If xrSubreport1.ReportSource.RowCount = 0 Then
				e.Cancel = True
				Return
			End If
			If (Not documentBuilt) Then
				Return
			End If
			If (Not sections.ContainsKey(customerID)) Then
				System.Diagnostics.Debug.WriteLine(customerID & " not found.") ' Not executed.
			Else
				xrLabel1.HeightF = 100 * GraphicsUnitConverter.Convert(sections(customerID), GraphicsUnit.Document, GraphicsUnit.Inch)
				CType(sender, DetailBand).HeightF = 100 * GraphicsUnitConverter.Convert(sections(customerID), GraphicsUnit.Document, GraphicsUnit.Inch)
			End If
		End Sub
	End Class
End Namespace
